parser.c

```diff
@@ -33,6 +33,7 @@ static int parser_read_char(struct kanshi_parser *parser) {
 		return ch;
 	}
 
+	errno = 0;
```

Comment:
I needed to do this because sometimes fgetc doesn't seem to reset errno, not sure what causes it.